beacon_chain/validator_client/api.nim

```
@@ -10,6 +10,133 @@ type
   ApiOperation = enum
     Success, Timeout, Failure, Interrupt
 
+  ApiNodeResponse*[T] = object
+    node*: BeaconNodeServerRef
+    data*: ApiResponse[T]
+
+  ApiResponseSeq*[T] = object
+    status*: ApiOperation
+    data*: seq[ApiNodeResponse[T]]
+
+template onceToAll*(vc: ValidatorClientRef, responseType: typedesc,
+                    timeout: Duration,
+                    body: untyped): ApiResponseSeq[responseType] =
+  var it {.inject.}: RestClientRef
+  type BodyType = typeof(body)
+
+  var timerFut =
+    if timeout != InfiniteDuration:
+      sleepAsync(timeout)
+    else:
+      nil
+
+  let onlineNodes =
+    try:
+      await vc.waitOnlineNodes(timerFut)
+      vc.onlineNodes()
+    except CancelledError:
+      var default: seq[BeaconNodeServerRef]
+      if not(isNil(timerFut)) and not(timerFut.finished()):
+        await timerFut.cancelAndWait()
+      default
+    except CatchableError:
+      # This case could not be happened.
```

If this truly can't happen, maybe log to check if it does? Are you sure enough about this to, e.g., `assert` or let it crash on a `Defect`?